Source/WarpX.cpp

```diff
@@ -2234,7 +2236,49 @@ WarpX::BuildBufferMasksInBox ( const amrex::Box tbx, amrex::IArrayBox &buffer_ma
 #endif
 }
 
-#ifdef WARPX_USE_PSATD
+amrex::Vector<amrex::Real> WarpX::getFornbergStencilCoefficients(const int n_order, const bool nodal)
+{
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(n_order % 2 == 0, "n_order must be even");
+
+    const int m = n_order / 2;
+    amrex::Vector<amrex::Real> coefs;
+    coefs.resize(m);
+
+    // There are closed-form formula for these coefficients, but they result in
+    // an overflow when evaluated numerically. One way to avoid the overflow is
+    // to calculate the coefficients by recurrence.
+
+    // Coefficients for nodal (centered) finite-difference approximation
+    if (nodal == true)
+    {
+       // First coefficient
+       coefs[0] = m * 2. / (m+1);
+       // Other coefficients by recurrence
+       for (int n = 1; n < m; n++)
+       {
+           coefs[n] = - (m-n) * 1. / (m+n+1) * coefs[n-1];
```

Comment:
Since this is not performance critical, consider maybe using `.at()` over `operator[]` for safety.

Reply:
OK, thanks for the suggestion. This function so far is called only at initialization, so the choice of the operator to access the elements of the `amrex::Vector` object is not critical for performance purposes, as you pointed out. I made the change in https://github.com/ECP-WarpX/WarpX/pull/2327/commits/7248ddf57467162db9f41a45b98cda5b6afba51e.